semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala

```diff
@@ -21,7 +22,11 @@ trait SymbolOps { self: SemanticdbOps =>
         if (sym.isTypeSkolem) return sym.deSkolemize.toSemantic
         if (sym.isSemanticdbLocal) return freshSymbol(sym.pos.toMeta.input)
 
-        val owner = sym.owner.toSemantic
+        val owner = {
+          val result = sym.owner.toSemantic
+          if (sym.isJavaStaticInnerClass) result.stripSuffix("#") + "."
+          else result
+        }
```

Comment:
Ooooohhh... The fact that owners become non-syntactic is super unfortunate. If we follow this route, we pretty much must have https://github.com/scalameta/scalameta/issues/1540 on our roadmap for RC1.

Reply:
Olaf and I had a chat, and the non-syntactic owner thing looks like a deal-breaker, since it: 1) will require us to give up `String.owner` and `Symbols.Global(owner, desc)`, which will be a pretty big deal internally, 2) will require a symbol table to construct a symbol from a string.